sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -1805,3 +1805,21 @@ class DynamicPartitionPruningV2SuiteAEOff extends DynamicPartitionPruningV2Suite
 
 class DynamicPartitionPruningV2SuiteAEOn extends DynamicPartitionPruningV2Suite
   with EnableAdaptiveExecutionSuite
+
+abstract class DynamicPartitionPruningV2FilterSuite
+    extends DynamicPartitionPruningDataSourceSuiteBase {
```

Comment:
shall we extend `DynamicPartitionPruningV2Suite` here? then we can save the `override protected def runAnalyzeColumnCommands: Boolean = false`, and catalog configs will be overwritten.

Reply:
Sounds good. I have a follow-up [here](https://github.com/apache/spark/pull/37643)